osquery/tables/events/linux/auditd_fim_events.cpp

```diff
@@ -0,0 +1,569 @@
+          configuration_.excluded_path_list.size());
+      configuration_.excluded_path_list.insert(
+          configuration_.excluded_path_list.end(),
+          solved_path_list.begin(),
+          solved_path_list.end());
+    }
+  }
+
+  if (root_key.find("show_accesses") != root_key.not_found()) {
+    auto key = root_key.get_child("show_accesses");
+    auto value = key.get_value<std::string>();
+
+    configuration_.show_accesses = (value == "true");
+  }
+}
+
+Status AuditdFimEventSubscriber::Callback(const ECRef& event_context,
+                                          const SCRef& subscription_context) {
+  std::vector<Row> emitted_row_list;
+  auto exit_status = ProcessEvents(emitted_row_list,
+                                   process_map_,
+                                   configuration_,
+                                   event_context->syscall_events);
+
+  for (auto& row : emitted_row_list) {
+    add(row);
+  }
+
+  return exit_status;
+}
+
+Status AuditdFimEventSubscriber::ProcessEvents(
+    std::vector<Row>& emitted_row_list,
+    AuditdFimProcessMap& process_map,
+    const AuditdFimConfiguration& configuration,
+    const std::vector<SyscallEvent>& syscall_event_list) noexcept {
+  // Configuration helpers
+  auto L_isPathIncluded = [&configuration](const std::string& path) -> bool {
+    return (std::find(configuration.included_path_list.begin(),
+                      configuration.included_path_list.end(),
+                      path) != configuration.included_path_list.end());
+  };
+
+  auto L_isPathExcluded = [&configuration](const std::string& path) -> bool {
+    return (std::find(configuration.excluded_path_list.begin(),
+                      configuration.excluded_path_list.end(),
+                      path) != configuration.excluded_path_list.end());
+  };
+
+  // Path normalization utility
+  auto L_normalizePath = [](const std::string& cwd,
+                            const std::string& path) -> std::string {
+    std::string translated_path;
+
+    // Remove the surrounding quotes, if any (they are only present if
+    // the path contains spaces)
+    if (path[0] == '"') {
+      translated_path = path.substr(1, path.size() - 2);
+    } else {
+      translated_path = path;
+    }
+
+    // Also use the working directory if the path is not absolute
+    if (translated_path[0] != '/') {
+      std::string translated_cwd;
+
+      if (cwd[0] == '"') {
+        translated_cwd = cwd.substr(1, cwd.size() - 2);
+      } else {
+        translated_cwd = cwd;
+      }
+
+      translated_path = translated_cwd + "/" + translated_path;
+    }
+
+    // Normalize the path; we could have used 'canonicalize()' but that
+    // accesses the file system and we don't want that (because it may
+    // spawn other events). It is also possible that by the time we
+    // inspect the files on disk the state that caused this event has
+    // changed.
+    namespace boostfs = boost::filesystem;
```

Comment:
Avoid aliasing within code, prefer at the top, where we commonly use `fs = boost::filesystem`.